File: app/src/main/java/com/funny/translation/js/JSManager.java
package com.funny.translation.js;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

public class JSManager {
    public static ArrayList<JSEngine> currentRunningJSList = new ArrayList<>();
    public static JSEngine currentRunningJSEngine;
    public static short currentRunningJSId = 99;

    public static short getJSEngineKind(){
        return currentRunningJSId;
    }

    public static void addJSEngine(JSEngine engine){
        if(!currentRunningJSList.contains(engine)){
            currentRunningJSList.add(engine);
        }
    }

    public static void clearAllEngines(){
        currentRunningJSList.clear();
    }

    @Nullable
    public static JSEngine getJSEngineById(int id){
        for (JSEngine jsEngine:currentRunningJSList){
            if (jsEngine.js.id==id)return jsEngine;
        }
        return null;
    }
}

File: base-core/src/main/java/com/funny/translation/js/JsManager.kt
package com.funny.translation.js

object JsManager {
    private val jsMap : HashMap<Int,JsEngine> = HashMap()
    var currentRunningJsEngine : JsEngine? = null

    fun getJsEngineById(id : Int) : JsEngine? = jsMap[id]
    fun addJSEngine(jsEngine: JsEngine){
        jsMap[jsEngine.getId()] = jsEngine
    }

    fun clear(){
        jsMap.clear()
    }
}